editor/src/components/canvas/canvas-strategies/flex-reorder-strategy.tsx

```diff
@@ -0,0 +1,160 @@
+import { MetadataUtils } from '../../../core/model/element-metadata-utils'
+import { ElementInstanceMetadataMap } from '../../../core/shared/element-template'
+import { offsetPoint } from '../../../core/shared/math-utils'
+import { ElementPath } from '../../../core/shared/project-file-types'
+import { reorderElement } from '../commands/reorder-element-command'
+import { CanvasStrategy } from './canvas-strategy-types'
+import {
+  CanvasPoint,
+  canvasPoint,
+  CanvasVector,
+  CanvasRectangle,
+} from '../../../core/shared/math-utils'
+import * as EP from '../../../core/shared/element-path'
+import { reverse, stripNulls } from '../../../core/shared/array-utils'
+
+export const flexReorderStrategy: CanvasStrategy = {
+  id: 'FLEX_REORDER',
+  name: 'Flex Reorder',
+  isApplicable: (canvasState, _interactionState, metadata) => {
+    if (canvasState.selectedElements.length == 1) {
+      return MetadataUtils.isParentYogaLayoutedContainerAndElementParticipatesInLayout(
+        canvasState.selectedElements[0],
+        metadata,
+      )
+    } else {
+      return false
+    }
+  },
+  controlsToRender: [], // Uses existing hooks in select-mode-hooks.tsx
+  fitness: (canvasState, interactionState, sessionState) => {
+    return flexReorderStrategy.isApplicable(
+      canvasState,
+      interactionState,
+      sessionState.startingMetadata,
+    ) &&
+      interactionState.interactionData.type === 'DRAG' &&
+      interactionState.activeControl.type === 'BOUNDING_AREA'
+      ? 1
+      : 0
+  },
+  apply: (canvasState, interactionState, sessionState) => {
+    if (
+      interactionState.interactionData.type !== 'DRAG' ||
+      interactionState.interactionData.drag == null
+    ) {
+      return []
+    }
+    const { selectedElements } = canvasState
+    const target = selectedElements[0]
+
+    const siblingsOfTarget = MetadataUtils.getSiblings(sessionState.startingMetadata, target).map(
+      (element) => element.elementPath,
+    )
+    const flexSiblingsOfTarget = siblingsOfTarget.filter((child) =>
```

Comment:
Might be worth having a test for the case where one of the siblings returns false for this conditional.

Reply:
Such a good point! I wrote it and realized that case has a bug! Fixing it now

Reply:
Fixed in 05f01159ddffc52b3a9d01283d7d027560861899